src/java/org/apache/cassandra/db/marshal/AbstractType.java

```diff
@@ -478,6 +493,21 @@ public final boolean isValueLengthFixed()
         return valueLengthIfFixed() != VARIABLE_LENGTH;
     }
 
+    public boolean allowsEmpty()
```

Comment:
Added

```
    /**
     * Defines if the type allows an empty set of bytes ({@code new byte[0]}) as valid input.  The {@link #validate(Object, ValueAccessor)}
     * and {@link #compose(Object, ValueAccessor)} methods must allow empty bytes when this returns true, and must reject empty bytes
     * when this is false.
     * <p/>
     * As of this writing, the main user of this API is for testing to know what types allow empty values and what types don't,
     * so that the data that gets generated understands when {@link ByteBufferUtil#EMPTY_BYTE_BUFFER} is allowed as valid data.
     */
```

Reply:
Thanks, the doc is perfectly clear.